nautobot/extras/models/models.py

```diff
@@ -822,13 +824,48 @@ def enqueue_job(cls, func, name, obj_type, user, *args, schedule=None, **kwargs)
         user: User object to link to the JobResult instance
         args: additional args passed to the callable
         schedule: Optional ScheduledJob instance to link to the JobResult
-        kwargs: additional kargs passed to the callable
+        celery_kwargs: Dictionary of options to pass as kwargs to Celery `apply_async` method
+        kwargs: additional kwargs passed to the `func` callable
         """
         job_result = cls.objects.create(name=name, obj_type=obj_type, user=user, job_id=uuid.uuid4(), schedule=schedule)
+        job_class = job_result.related_object
 
+        # Explicitly pass along the PK of the JobResult
         kwargs["job_result_pk"] = job_result.pk
 
-        func.apply_async(args=args, kwargs=kwargs, task_id=str(job_result.job_id))
+        # Prepare kwargs that will be sent to Celery
+        if celery_kwargs is None:
+            celery_kwargs = {}
+
+        # Set the specialized Singleton options
+        once_default_run_job = "nautobot.extras.jobs.run_job"
+        once_default_keys = getattr(job_class, "singleton_keys", None) or ["data", "commit"]
+        is_singleton = getattr(job_class, "singleton", False)
+        celery_kwargs["singleton"] = is_singleton
+
+        # If the default `run_job` function is being used, default to doing Singleton on its
+        # argument keys, unless they are explicitly passed in by the caller.
+        if func.name == once_default_run_job and "once" not in celery_kwargs:
```

Comment:
Instead of checking `func.name` here, could we actually do something like:

```python
from nautobot.extras.jobs import run_job

...

if func is run_job and "once" not in celery_kwargs:
```